sklearn/inspection/_permutation_importance.py

```diff
@@ -28,17 +28,19 @@ def _safe_column_indexing(X, col_idx):
 def _calculate_permutation_scores(estimator, X, y, col_idx, random_state,
                                   n_repeats, scorer):
     """Calculate score when `col_idx` is permuted."""
-    original_feature = _safe_column_indexing(X, col_idx).copy()
-    temp = original_feature.copy()
+    random_state = check_random_state(random_state)
 
+    # Work on a copy of X to to ensure thread-safety in case of threading
+    # based parallelism:
+    X_permuted = X.copy()
```

Comment:
Note for the reviewers: the fact that we always make a copy here also fixes the issue with read-only memmaps as reported in #15810.

Reply:
Just by chance, you are not aware of a way to avoid to re-allocating the full dataframe, and instead make a view for most columns except for the one we want to change inplace?

For instance of one makes a slice of a dataframe, and then tries to modify to a column, pandas will raise a warning about a view being modified, but I'm not sure if it will actually change the original dataframe inplace or not in this case...

Reply:
It depends on the internal block structure of the dataframe but this is considered private API and is likely to change in future versions of pandas. I would rather stay safe for now.

Reply:
In the future, I can see this being possible when pandas switches to using a columnar data structure to hold its data as detailed in [the pandas roadmap](https://dev.pandas.io/docs/development/roadmap.html#block-manager-rewrite).